direct-hs/src/Web/Direct/Client.hs

```diff
@@ -129,35 +133,28 @@ findTalkRoom tid client = do
 ----------------------------------------------------------------
 
 leaveTalkRoom :: Client -> TalkId -> IO (Either Exception ())
-leaveTalkRoom client tid = do
-    mtalk <- L.find ((tid ==) . talkId) <$> getTalkRooms client
-    case mtalk of
-        Nothing   -> return $ Left InvalidTalkId
-        Just talk -> do
-            mme <- getMe client
-            case mme of
-                Nothing -> fail "Assertion failure: `getMe` returns `Nothing`"
-                Just me -> do
-                    deleteTalker (clientRpcClient client) talk me
-                    return $ Right ()
+leaveTalkRoom client tid = runExceptT $ do
+    talk <- failWith InvalidTalkId =<< liftIO (findTalkRoom tid client)
+    mme  <- liftIO $ getMe client
+    case mme of
+        Just me -> do
+            result <- liftIO $ deleteTalker (clientRpcClient client) talk me
```

Comment:
`ExceptT $ deleteTalker (clientRpcClient client) talk user` instead of `liftIO` would delete the `case` expression below.

Reply:
Oh, I see! :boom:
I rewritten it with `ExceptT`.